context/src/main/java/io/grpc/Context.java

```diff
@@ -881,8 +881,10 @@ public String toString() {
      * Implements {@link io.grpc.Context#attach}.
      *
      * @param toAttach the context to be attached
+     * @return A {@link Context} that should be passed back into {@link #detach(Context, Context)}
+     *        as the {@code toRestore} parameter.
      */
-    public abstract void attach(Context toAttach);
+    public abstract Context attach(Context toAttach);
```

Comment:
Performed a simple test and verified that changing a void return type to a non avoid return type is not binary compatible.